helix-core/src/test/java/org/apache/helix/integration/task/TestDeleteJobFromJobQueue.java

```diff
@@ -69,11 +69,18 @@ public void testForceDeleteJobFromJobQueue() throws InterruptedException {
     Assert
         .assertNotNull(_driver.getJobContext(TaskUtil.getNamespacedJobName(jobQueueName, "job2")));
 
-    // The following force delete for the job should go through without getting an exception
+    // force deletion can have Exception thrown as controller is writing to propertystore path too.
+    // https://github.com/apache/helix/issues/1406
+
+    // Thus, we stop pipeline to make sure there is not such race condition.
+    _gSetupTool.getClusterManagementTool().enableCluster(CLUSTER_NAME, false);
+    Thread.sleep(3000); // note this sleep is critical as it would take time for controller to stop
```

Comment:
I think you also emphasized that we shall avoid using Thread.sleep() in the tests. 3-second sleep is not guaranteed to work.
Can we use verifier with a timeout to wait until the pipeline stopped?

Reply:
How do we verify the pipeline stops processing event? Note, you can't just rely on pause signal node is in the Zookeeper. 

This is similar to waitTillVerify(). It seems to me there is not good way to wait for some signal.

Reply:
Is it better to retry `deleteJob` then? Disable the cluster and retry until no exception.